New layout nodes

mkaring opened the pull request:
Added two new layout nodes

mkaring committed:
NiftyLayout - First step of layout change according to discussion

mkaring committed:
feat: Added new padding layout node

mkaring committed:
feat: Added new fixed size node

mkaring committed:
refactor: improved padding layout node

void256 commented:
The pull request now has conflicts. Can you check please? It's too late for me right now =)

mkaring committed:
merge: Merged latest changes and resolved conflicts

Merge branch 'new-node-tree-heavy-wip' of https://github.com/void256/nifty-gui into 2.0

Conflicts:
	nifty/src/main/java/de/lessvoid/nifty/Nifty.java
	nifty/src/main/java/de/lessvoid/nifty/NiftyLayout.java
	nifty/src/main/java/de/lessvoid/nifty/node/StackLayoutNodeImpl.java
	nifty/src/main/java/de/lessvoid/nifty/node/UniformStackLayoutNodeImpl.java

mkaring committed:
feat: Added accessors for layout nodes

mkaring commented:
I did the merge and updated the pull request.
